### PR TITLE
Add str_remove_before, str_remove_after, str_extract_before, str_extract_after, str_replace_before, str_replace_after

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

Supported

- str_remove_before
- str_remove_after
- str_replace_before
- str_replace_after
- str_extract_before
- str_extract_after

# Checklist
Make sure you have performed the following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/string_operation.R [hunk 1]
```diff
@@ -909,6 +909,63 @@ str_remove_emoji <- function(column, position = "any"){
...
+}
+
+#'Function to extract text after the separator.
+#'
+#'export
+str_extract_after <- function(column, sep = "\\,", include_sep = TRUE){
+  if (include_sep){
+    stringr::str_extract(column, stringr::str_c(sep, ".*"))
+  } else {
+    #(?<=あ)(.*)
```
Comment (hideaki) [inline R/string_operation.R:930]: Please remove or explain the above line.

File: R/string_operation.R [hunk 2]
```diff
@@ -909,6 +909,62 @@ str_remove_emoji <- function(column, position = "any"){
   stringi::stri_replace_all(column, regex = regexp, "")
 }
 
+#'Function to extract text before the separator.
+#'
+#'export
+str_extract_before <- function(column, sep = "\\,", include_sep = TRUE) {
```
Comment (hideaki) [inline R/string_operation.R:915]: Is include_sep=TRUE really the more common usage?

File: R/string_operation.R [hunk 3]
```diff
@@ -909,6 +909,62 @@ str_remove_emoji <- function(column, position = "any"){
...
+    stringr::str_extract(column, stringr::str_c(".*", sep))
+  } else {
+    stringr::str_extract(column, stringr::str_c("(.*)(?=", sep, ")"))
+  }
+}
+
+#'Function to extract text after the separator.
+#'
+#'export
+str_extract_after <- function(column, sep = "\\,", include_sep = TRUE){
```
Comment (hideaki) [inline R/string_operation.R:926]: Is include_sep=TRUE really the more common usage?

File: R/string_operation.R [hunk 4]
```diff
@@ -909,6 +909,62 @@ str_remove_emoji <- function(column, position = "any"){
...
+  if (include_sep){
+    stringr::str_extract(column, stringr::str_c(sep, ".*"))
+  } else {
+    stringr::str_extract(column, stringr::str_c("(?<=", sep, ")(.*)"))
+  }
+}
+#'Function to replace text before the separator.
+#'
+#'export
+str_replace_before <- function(column, sep = ",", rep = "", include_sep = TRUE) {
```
Comment (hideaki) [inline R/string_operation.R:936]: I think whichever would work, but only this function's sep is not escaped by `\\`.

File: R/string_operation.R (hunk 1 above)
Comment (hidekoji) [reply R/string_operation.R]: Removed it.

File: R/string_operation.R (hunk 4 above)
Comment (hidekoji) [reply R/string_operation.R]: added `\\` for consistency.

File: R/string_operation.R (hunk 2 above)
Comment (hidekoji) [reply R/string_operation.R]: Changed the default value as suggested.

File: R/string_operation.R (hunk 3 above)
Comment (hidekoji) [reply R/string_operation.R]: Changed the default value as suggested.

**Pull request merged**